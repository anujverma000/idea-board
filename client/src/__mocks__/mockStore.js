import configureStore from 'redux-mock-store'
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState= {
    idea: {
        list: []
    }
}
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
store.replaceReducer(rootReducer);

export default store;