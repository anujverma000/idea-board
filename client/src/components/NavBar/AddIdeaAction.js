import React from "react";
import { connect } from 'react-redux';
import styled from "styled-components";
import { createIdea } from '../../actions/ideaActions';

const AddIdeaAction = ({ createNewIdea }) => {
    return(
        <Wrapper onClick={createNewIdea} >Add new idea</Wrapper>
    )
}


/**
 *  STYLES
 */
const Wrapper = styled.button`
    color: ${props => props.theme.textColorPrimary};
    padding: 16px 24px;
    font-size: 16px;
    border-radius: 4px;
    cursor: pointer;
    box-sizing: border-box;
    min-width: 150px;
    text-align: center;
    border: 1px solid ${props => props.theme.textColorPrimary};
    background-color: ${props => props.theme.backgroundColor};
    &:hover{
        transition: .3s;
        color: ${props => props.theme.textColorInvert};
        background-color: ${props => props.theme.primaryColorLight};
    }
    &:active{
        color: ${props => props.theme.primaryColor};
    }
`;

const mapDispatchToProps = dispatch => ({
    createNewIdea: () =>  dispatch(createIdea())
});

export default connect(null ,mapDispatchToProps)(AddIdeaAction)