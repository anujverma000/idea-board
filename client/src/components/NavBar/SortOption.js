import React, { useState } from "react";
import styled from "styled-components";

const SortOption = ({ field, label, sort }) => {
  const [ sortOrder ,setSortOrder ] = useState('');

  const handleSort = () => {
    let type = sortOrder === 'desc' ? 'asc': 'desc'
    setSortOrder(type);
    sort({field, type})
  }
  return(
    <Option onClick={handleSort}>
      { label } 
      { sortOrder === '' && <DefaultSort/> }
      { sortOrder === 'asc' && <Up/> }
      { sortOrder === 'desc' && <Down/> }
    </Option>
  )
}

const Option = styled.div`
  font-size: 16px;
  text-align: center;
  cursor: pointer;
  padding: 12px;
  &:hover{
    background-color: ${props => props.theme.borderColor};  
  }
  border-bottom: 1px solid ${props => props.theme.borderColor};
  &:last-child{
    border-bottom: none;
  }
`
const DefaultSort = styled.div`
  width: 0;
  height: 0;
  float: right;
  margin: 4px;
  border-style: solid;
  border-width: 0 5px 8.7px 5px;
  border-color: transparent transparent ${props => props.theme.shadowColor} transparent;
`;

const Up = styled(DefaultSort)`
  border-width: 0 5px 8.7px 5px;
  border-color: transparent transparent ${props => props.theme.primaryColor} transparent;
`;

const Down = styled(DefaultSort)`
  border-width: 8.7px 5px 0 5px;
  border-color: ${props => props.theme.primaryColor} transparent transparent transparent;
`;
export default SortOption;