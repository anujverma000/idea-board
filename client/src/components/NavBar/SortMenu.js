import React, { useState, useRef, useEffect } from "react";
import SortOption from './SortOption'
import styled from "styled-components";

const SortMenu = ({ sort }) => {
  const [ displayMenu ,setDisplayMenu ] = useState(false);
  const node = useRef();

  useEffect(() => {
    document.addEventListener('mouseup', handleClickOutside)
    return () => {
      document.removeEventListener('mouseup', handleClickOutside)
    }
  },[])
  
  const handleClickOutside = e => {
    if (node.current && !node.current.contains(e.target)) {
      setDisplayMenu(false)
    }
  }

  return(
    <Wrapper ref={ node } onClick={() => setDisplayMenu(!displayMenu) }>
      <Sort>Sort</Sort>
      { displayMenu &&
        <OptionsMenu>
          <SortOption field='title' sort={sort} label='Title'/>
          <SortOption field='date' sort={sort} label='Date'/>
        </OptionsMenu>
      }
    </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.div`
  position: relative;
`;
const Sort = styled.div`
  color: ${props => props.theme.textColorPrimary};
  padding: 16px 24px;
  font-size: 16px;
  border-radius: 4px;
  cursor: pointer;
  margin: 0 16px;
  min-width: 150px;
  box-sizing: border-box;
  text-align: center;
  border: 1px solid ${props => props.theme.textColorPrimary};
  background-color: ${props => props.theme.backgroundColor};
  &:hover{
      transition: .3s;
      color: ${props => props.theme.textColorInvert};
      background-color: ${props => props.theme.primaryColorLight};
  }
  &:active{
      color: ${props => props.theme.primaryColor};
  }
`;

const OptionsMenu = styled.div`
  background-color: ${props => props.theme.backgroundColor};
  display: flex;
  position: absolute;
  top: 60px;
  flex-direction: column;
  min-width: 200px;
  border: 1px solid ${props => props.theme.borderColor};
  border-radius: 4px;
  z-index: 3;
  box-shadow: 0 0 12px 4px ${props => props.theme.shadowColor};
`;

export default SortMenu;