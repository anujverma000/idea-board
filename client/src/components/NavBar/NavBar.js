import React from "react";
import styled from "styled-components";
import SortMenu from "./SortMenu";
import AddIdeaAction from "./AddIdeaAction";

const NavBar = ({sort}) => {
  return(
    <Wrapper>
      <SortMenu sort = {sort}/>
      <AddIdeaAction/>
    </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.nav`
  padding: 24px;
  display: flex;
  justify-content: flex-end;
`;

export default NavBar;