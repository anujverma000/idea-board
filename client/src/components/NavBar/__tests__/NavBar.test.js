import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar'
import mockStore from "../../../__mocks__/mockStore";
import { Provider } from "react-redux";

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={mockStore}><NavBar /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
