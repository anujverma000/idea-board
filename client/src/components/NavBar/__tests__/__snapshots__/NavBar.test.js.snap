// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="sc-htoDjs gbJRkw"
>
  <div
    className="sc-ifAKCX bqyzQp"
    onClick={[Function]}
  >
    <div
      className="sc-EHOje dFQdKu"
    >
      Sort
    </div>
  </div>
  <button
    className="sc-gzVnrw kuvISq"
    onClick={[Function]}
  >
    Add new idea
  </button>
</nav>
`;
